common/app/views/fragments/header.scala.html

```diff
@@ -9,7 +9,7 @@
     @fragments.newHeader()
 } else {
     <header id="header"
-            class="l-header u-cf @if(page.metadata.hasSlimHeader) {l-header--is-slim l-header--no-navigation} js-header"
+            class="l-header u-cf @if(page.metadata.hasSlimHeader) {l-header--is-slim l-header--no-navigation} @if(mvt.ABGalleryRedesignVariant.isParticipating && page.metadata.contentType.toLowerCase == "gallery"){l-header--is-slim l-header--no-navigation l-header--new-gallery} js-header"
```

Comment:
Changed this to `shouldShow` method which includes the check of whether you are on a gallery or not.